### PR TITLE
[DOC] Replace `sphinx-panels` with `sphinx-design`

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Replaces the `sphinx-panels` dependency with `sphinx-design` #3370.

The cards/panels are slightly smaller than before, and the button colour has changed since the default grey bugged out on darkmode for me with the new package.

### CODE REVIEW

**Review by TonyBagnall — APPROVED** [review]

**Review by fkiraly — CHANGES_REQUESTED** [review]
Why is this needed? What does it improve?

**Comment by MatthewMiddlehurst** [comment]
See #3370. You gave it a thumbs up, so I assumed you agreed with the change.

**Comment by RNKuhns** [comment]
@MatthewMiddlehurst I'll give this a review in the next day.

@fkiraly the `sphinx-panels` package that has been used in the `sktime` documentation is no longer maintained. `sphinx-design` provides similar functionality and is suggested as an alternative by [sphinx-panels](https://sphinx-panels.readthedocs.io/en/latest/index.html). Plus `sphinx-design` looks nice (IMHO).

**Comment by fkiraly** [comment]
ah yes, I remember, thanks. Was confused by the PR text which said something about panels being smaller, which struck me odd as a reason for a change, but you have to follow the link to #3370. Please consider this unblocked (but not approved)

**Comment by fkiraly** [comment]
any chance to get this working using `sphinx-design`?
https://github.com/sktime/sktime/pull/3211

Using `sphinx-panels`, it was impossible to get two buttons on a panel and have both work.

**Comment by achieveordie** [comment]
LGTM

**Comment by MatthewMiddlehurst** [comment]
@RNKuhns do you still intend to review this? Not rushing you to do so if you are.

File: docs/source/index.rst [hunk 1]
```diff
@@ -47,103 +47,131 @@ Contents
...
-    User Guide
-    ^^^^^
+            Get Started
 
-    Find user documentation.
+    .. grid-item-card::
+        :text-align: center
 
-    +++
+        User Guide
```
Comment (RNKuhns) [inline docs/source/index.rst:74]: This is really pointing to the broader category of "User Documentation". Could consider changing these labels to "User Documentation" from "User Guide". But this PR does successfully convert to sphinx-design and retain the original label.

**Review by RNKuhns — APPROVED** [review]
@MatthewMiddlehurst this looks good to me and the new buttons point to the same places as the old buttons. I think we should be able to merge this.

I did make one comment on the label for the "User Guide" on the index. It is really pointing to the "User Documentation". But you maintained the prior label, so maybe we shouldn't change that as part of this PR.

**Pull request merged**